main.tf

```
@@ -223,7 +223,7 @@ resource "aws_batch_job_queue" "this" {
   name                  = each.value.name
   state                 = each.value.state
   priority              = each.value.priority
-  scheduling_policy_arn = try(each.value.scheduling_policy_arn, aws_batch_scheduling_policy.this[each.key].arn)
+  scheduling_policy_arn = lookup(each.value, "attach_scheduling_policy", true) ? try(each.value.scheduling_policy_arn, aws_batch_scheduling_policy.this[each.key].arn) : null
```

lets change this to 
```suggestion
  scheduling_policy_arn = try(each.value.create_scheduling_policy, true) ? aws_batch_scheduling_policy.this[each.key].arn : try(each.value.scheduling_policy_arn, null)
```

then down on line 237 (can't comment on lines not in diff) - lets change line 237 to

```hcl
  for_each = { for k, v in var.job_queues : k => v if var.create && var.create_job_queues && try(v.create_scheduling_policy, true) }
```

---

Changed, I didn't actually know you could use values in the for_each